cyberdrop_dl/managers/path_manager.py

```diff
@@ -110,6 +116,22 @@ def _set_output_filenames(self) -> None:
             internal_name = f"{model_name.replace('_log','')}_log"
             setattr(self, internal_name, self.log_folder / getattr(log_settings_config, model_name))
 
+    def _delete_logs_and_folders(self):
+        log_settings_config = self.manager.config_manager.settings_data.logs
+        log_files: dict[str, Path] = log_settings_config.model_dump()
+        if self.manager.config_manager.settings_data.logs.max_num_logs:
+            all_logs = list(Path(self.log_folder).rglob("*.*"))
+            all_logs_sorted = sorted(all_logs, key=lambda f: Path(f).stat().st_ctime, reverse=True)
+            max_prev_logs = self.manager.config_manager.settings_data.logs.max_num_logs - 1
+            for model_name, log_file in log_files.items():
+                if model_name not in self._logs_model_names:
+                    continue
+                for file in list(filter(lambda x: str(x).find(log_file.stem) != -1, all_logs_sorted))[max_prev_logs:]:
+                    file.unlink(missing_ok=True)
+        for curr, _, _ in os.walk(self.log_folder, topdown=False):
+            if not os.listdir(curr):
+                Path(curr).rmdir()
+
```

Comment:
Call `purge_dir_tree` instead to also delete any possible empty log file

```suggestion
         purge_dir_tree(self.log_folder)
```